packages/editor/src/components/post-schedule/index.js

```diff
@@ -45,6 +53,7 @@ export default function PostSchedule( { onClose } ) {
 				after: getDayOfTheMonth( previewedMonth ),
 				before: getDayOfTheMonth( previewedMonth, false ),
 				exclude: [ select( editorStore ).getCurrentPostId() ],
+				per_page: getDaysInMonth( previewedMonth ),
 			} ),
 		[ previewedMonth, postType ]
 	);
```

Comment:
It's not related to this PR but would you mind removing `type` and `title` from `events` below? They're not used. Each event object only needs to contain `date`.

https://github.com/WordPress/gutenberg/blob/401be48d984ff0959598417e5bd94d24cdfe364b/packages/components/src/date-time/types.ts#L26-L31

Reply:
No problem! Took care of that in 556b20ed2e